core/sail/fts/lucene-spin/src/main/java/org/eclipse/rdf4j/lucene/spin/LuceneSpinSail.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2017 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+package org.eclipse.rdf4j.lucene.spin;
+
+import java.io.File;
+import java.util.Properties;
+import org.eclipse.rdf4j.sail.NotifyingSail;
+import org.eclipse.rdf4j.sail.NotifyingSailConnection;
+import org.eclipse.rdf4j.sail.SailException;
+import org.eclipse.rdf4j.sail.evaluation.TupleFunctionEvaluationMode;
+import org.eclipse.rdf4j.sail.helpers.NotifyingSailWrapper;
+import org.eclipse.rdf4j.sail.lucene.LuceneSail;
+import org.eclipse.rdf4j.sail.lucene.SearchIndex;
+import org.eclipse.rdf4j.sail.lucene.SearchIndexQueryContextInitializer;
+import org.eclipse.rdf4j.sail.lucene.util.SearchIndexUtils;
+import org.eclipse.rdf4j.sail.spin.SpinSail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Activates support of {@link SearchIndex} feature inside {@link SpinSail} and manages the index during
+ * statements adding/removing. Technically this sail binds {@link SearchIndex} using
+ * .addQueryContextInitializer and wraps connection from baseSail by wrapped which modify SearchIndex.
+ *
+ * @author github.com/jgrzebyta
+ */
+public class LuceneSpinSail extends NotifyingSailWrapper {
+
+	private final Logger log = LoggerFactory.getLogger(LuceneSpinSail.class);
+
+	private SearchIndex si;
+
+	private Properties parameters = new Properties();
+
+	public LuceneSpinSail() {
+	}
+
+	public LuceneSpinSail(NotifyingSail baseSail) {
```

Comment:
The initialize method presumes (no instanceof check) that baseSail is a direct instance of SpinSail. Is it better to enforce that in this constructor?